src/gen/enums.rs

```diff
@@ -89,34 +86,32 @@ pub fn generate_enum(obj: &Enum, type_helper: &dyn TypeHelperRenderer) -> dart::
 
         for (index, obj) in obj.variants().iter().enumerate() {
             for f in obj.fields() {
-                // make sure all our field types are added to the includes
                 type_helper.include_once_check(&f.as_codetype().canonical_name(), &f.as_type());
             }
             variants.push(quote!{
                 class $(class_name(obj.name()))$cls_name extends $cls_name {
-
                     $(for field in obj.fields() => final $(&field.as_type().as_renderable().render_type(&field.as_type(), type_helper)) $(var_name(field.name()));  )
 
                     $(class_name(obj.name()))$cls_name._($(for field in obj.fields() => this.$(var_name(field.name())), ));
 
-                    static LiftRetVal<$(class_name(obj.name()))$cls_name> read(Api api, Uint8List buf) {
+                    static LiftRetVal<$(class_name(obj.name()))$cls_name> read( Uint8List buf) {
                         int new_offset = buf.offsetInBytes;
 
                         $(for f in obj.fields() =>
-                            final $(var_name(f.name()))_lifted = $(f.as_type().as_codetype().ffi_converter_name()).read(api, Uint8List.view(buf.buffer, new_offset));
+                            final $(var_name(f.name()))_lifted = $(f.as_type().as_codetype().ffi_converter_name()).read(Uint8List.view(buf.buffer, new_offset));
                             final $(var_name(f.name())) = $(var_name(f.name()))_lifted.value;
                             new_offset += $(var_name(f.name()))_lifted.bytesRead;
                         )
                         return LiftRetVal($(class_name(obj.name()))$cls_name._(
                             $(for f in obj.fields() => $(var_name(f.name())),)
-                        ), new_offset);
+                        ), new_offset - buf.offsetInBytes);
```

Comment:
can you elaborate on this change? It seems odd to wrong ... considering that we start out with `int new_offset = buf.offsetInBytes;` in L98, why would we add stuff and then substract that same offset again? Did you change the meaning of the return value?